tests/README.md

```diff
@@ -0,0 +1,90 @@
+
+1. Test that run before installation on any machine
+    - test of os-check (centos check and kernel)
+    - test first installation of component that doesn't need kubernetes (node_exporter)
+    - test of the complete installation it-self
+
+2. Test that run after the installation of kubernetes only (not the entire set of MetalK8s):
+    - first installation of elasticsearch
+
+3. Test that could be run at each point of time after MetalK8s installation (non-destructive):
+    - test of prometheus monitoring
+
+4. Test that meant to be run after installation and before production (conformance test)
+    - test kubernetes object creation
+
+5. Test that need to run on non-production environnement (CI test)
+    - test of failover
+
+6. Test that could run with only mocked environement:
+    - inventory check
+
+7. Unittesting of ansible roles: molecule
+
+```
+pytest --inventory <path_to_inventory> --install --conformance --go-destructive
+```
+
+Command line arguments:
+
+* `--install` trigger installation test and fixture (handles `.1` and `.2` test category)
+
+* `--conformance` enable conformance test (handles `.4`)
+
+* `--go-destructive` enable full testing, including test of failover (handles `.5`)
+
+No args on pytest mean only non-destructive test (handle case `.3` and `.6`)
+
+This represents e2e/integration testing of MetalK8s. Might be better to
+have unittesting as a separate test-suite (Molecule, different python scripts in the repo)
+
+
+#### How to order installation test
+We need ordering between `.1` tests and test that have an installation prerequisite.
+For this we could use the fact that pytest will tend to reduce amount of feature
+
+```
+def install_metal_k8s(node_exporter_install_fixture, pre_storage_install_fixture)
+    if install_flag:
+        run_ansible('playbooks/deploy.yml')
+```
+Thus, pytest might run tests that need node_exporter_install_fixture and pre_storage_install_fixtures.
+CON: installation is done during fixture, not during test.
+
+
+
+*Proposal*: Handle everything in pytest and handles `.7` in a different test-suite (and other unittesting).
+
+*Alternative proposal*: Handle pre-installation in one pytest test-suite.
+    Handle post-installation in another test-suite. A final one for `7.` and different unit-testing
+
+
+
+# Implementation details.
+
+### Running ansible
+Recommendation: Use https://github.com/pytest-dev/pytest-bdd, when possible (important tests for example)
+
+pytest will need to run ansible. We have two choices here:
+
+
+* subprocess.check_call(‘ansible-playbook’):
+    PRO:
+    - better resilience to API changes
+    CON:
+    - cannot extract internals of ansible (registered variable, inventory parsing)
+
+* wrap ansible API (50- 100 lines of codes)
+    PRO:
+    - can make assertion about ansible internals
+    CON:
+    - less stable to ansible upgrade
+
+### Accessing kubernetes services
+
+* telepresence ?
+* multiprocess.run('kubectl port-forward') ?
```

Comment:
Right now we document service access through `kubectl proxy`, so that's what tests should do.

Reply:
https://github.com/pytest-dev/pytest-xprocess could be useful here?

Reply:
Or https://github.com/pytest-dev/pytest-services